### PR TITLE
API/ENH: More verbose suspenders

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: bluesky/suspenders.py [hunk 1]
```diff
@@ -162,13 +163,10 @@ def get_futures(self):
         -------
         futs : list
             List of futures to wait on
-
-        msg : str
-            String explaining why the suspender is tripped
         '''
         if not self.tripped:
-            return [], ''
-        return [self.__make_event().wait()], self.get_justification()
```
Comment (danielballan) [inline bluesky/suspenders.py]: Here the API is reverted to what it was. I gather that this was a quick fix and not an especially intentional API choice. But feel free to argue why we shouldn't just call `get_justification()` separately, @tacaswell.

**Comment by tacaswell** [comment]
:-1:  The hacked API here is `get_justification` which the RE does not know about on master.

Getting the futures and the message needs to be atomic (as the reasons the suspender might report it is suspending may be different by the time we call `get_justification()`. )

File: bluesky/run_engine.py [hunk 2]
```diff
@@ -640,7 +643,8 @@ def remove_suspender(self, suspender):
             suspender.remove()
         self._suspenders.discard(suspender)
 
-    def request_suspend(self, fut, *, pre_plan=None, post_plan=None):
+    def request_suspend(self, fut, *, pre_plan=None, post_plan=None,
+                        justification=None):
```
Comment (tacaswell) [inline bluesky/run_engine.py]: :+1: on this API

File: bluesky/run_engine.py [hunk 3]
```diff
@@ -653,15 +657,20 @@ def request_suspend(self, fut, *, pre_plan=None, post_plan=None):
...
+        justification : str, optional
+            explanation of why the suspension has been requested
         """
         if not self.resumable:
             print("No checkpoint; cannot suspend. Aborting...")
             self._exception = FailedPause()
         else:
             print("Suspending....To get prompt hit Ctrl-C twice to pause.")
+            if justification is None:
+                print("No justification for this suspension was given.")
```
Comment (tacaswell) [inline bluesky/run_engine.py]: if it is optional do not include passive aggressive commentary :wink:

**Comment by danielballan** [comment]
> Getting the futures and the message needs to be atomic

Got it. That makes sense. I reverted by changed to `get_futures` and force-pushed.

**Pull request merged**